keyvalue/clients/src/main/java/tachyon/client/keyvalue/LinearProbingIndex.java

```diff
@@ -79,9 +81,16 @@ public static LinearProbingIndex createEmptyIndex() {
    * @return an instance of linear probing index
    */
   public static LinearProbingIndex loadFromByteArray(ByteBuffer buffer) {
-    int numBuckets = buffer.limit() / BUCKET_SIZE_BYTES;
-    // TODO(binfan): fix the key count which is wrong now, see TACHYON-1555
-    return new LinearProbingIndex(buffer, numBuckets, 0);
+    int numBuckets = 0;
+    int keyCount = 0;
+    for (int offset = 0, limit = buffer.limit(); offset < limit; offset += BUCKET_SIZE_BYTES) {
```

Comment:
I see your point, but I'm concerned that it's unusual to initialize two variables but only iterate one of them, so this might be less readable by being unnecessarily nonstandard. I'd be ok with getting overruled on this if anyone else feels strongly.